EvidenceApi/V1/Gateways/ResidentsGateway.cs

```diff
@@ -61,6 +62,22 @@ public Resident FindResident(Resident request)
                     r.PhoneNumber == request.PhoneNumber);
         }
 
+
+        public Resident FindResidentByGroupId(ResidentSearchQuery request)
```

Comment:
We might only need to receive a `groupId` (Guid) instead of a request object of type `ResidentSearchQuery`. There is a bit more context about this in the following comment.

Reply:
I think this was thought out this way in order to include this search in what's already in place (see the considerations in the ticket: https://hackney.atlassian.net/browse/DOC-1397) This way, we don't need to have a different endpoint or change the input of the one currently in place.
Since the other two parameters of the request object are strings, we can just leave them empty when we need to retrieve the resident by groupId.